server/uploader.js

```diff
@@ -90,44 +102,50 @@ module.exports = {
         var xl = parseInt(linePart[1]);
         var xr = parseInt(linePart[2]);
         var val = parseFloat(linePart[3]);
-        //console.log(seg);
+        lastChrXr[chName] = xr;
```

Comment:
It seems that you are using lastChrXr to replace lastxr. lastxr can be removed then.